docs/src/design/proposals/APIRedesign_PeteRichards.md

```diff
@@ -0,0 +1,246 @@
+
+As an example, registering a specific type of model provider in angular would look like:
+
+```javascript
+mct.provider('model', modelProvider() { /* implementation */});
+
+mct.config(['modelProvider', function (modelProvider) {
+    modelProvider.providers.push(RootModelProvider);
+}]);
+```
+
+## Less angular: only for views
+
+The other reason to move away from angular DI is angular DI makes it impossible to take one of our common services (say the persistence service) and start running it in a worker without loading the entire angular framework.  Webworkers should not have module loading functionality than the rest of the platform.
+
+If we wish to use less angular, I would recommend discontinuing use of all angular components that are not view related-- services, factories, $http, etc, and implementing them in our own paradigm.  Otherwise, we end up with layered interfaces-- one of the goals we would like to avoid.
+
+
+# Standard packaging and build system
+
+Standardize the packaging and build system, and completely separate the core platform from deployments.  Prescribe a starting point for deployments, but allow flexibility.
+
+## Use systemjs for module loading
+
+Allow developers to use whatever module loading system they'd like to use, while still supporting all standard cases.  We should also use this system for loading assets (css, scss, html templates), which makes it easier to implement a single file deployment using standard build tooling.
+
+## Use gulp or grunt for standard tooling
+
+Using gulp or grunt as a task runner would bring us in line with standard web developer workflows and help standardize rendering, deployment, and packaging.  Additional tools can be added to the workflow at low cost, simplifying the set up of developer environments.
+
+Gulp and grunt provide useful developer tooling such as live reload, automatic scss/less/etc compiliation, and ease of extensibility for standard production build processes.  They're key in decoupling code.
+
+## Package openmctweb as single versioned file.
+
+Deployments should depend on a specific version of openmctweb, but otherwise be allowed to have their own deployment and development toolsets.
+
+Customizations and deployments of openmctweb should not use the same build tooling as the core platform; instead they should be free to use their own build tools as they wish.  (We would provide a template for an application, based on our experience with warp-for-rp and vista)
+
+Installation and utilization of openmctweb should be as simple as downloading the js file, including it in your own html page, and then initializing an app and running it.  If a developer would prefer, they could use bower or npm to handle installation. 
+
+Then, if we're using imperative methods for extending the application we can use the following for basic customization:
+
+```html
+<script src="//localhost/openmctweb.js"></script>
```

Comment:
Want to highlight the importance of this style of deployment: you can explain and demonstrate things in gists and your "getting started" guides can cut out acres of bootstrapping and just let someone start programming in a single html file. (or even in a jsfiddle).

This also has the benefit of making bug reports incredibly easy to [report](https://github.com/angular/angular.js/issues/12795), [reproduce](https://gist.github.com/VWoeltjen/c0d9b9bb182e96725891), and [resolve](https://github.com/gkalpak/angular.js/commit/d5c981ba171fa24533196d0ad3ad8133abfd4078).